packages/server/lib/controllers/oauth.controller.ts

```diff
@@ -1251,6 +1276,26 @@ class OAuthController {
                     environmentId: environment.id,
                     accountId: account.id
                 });
+                if (!updatedConnection) {
+                    await logCtx.error('Failed to create connection');
+                    await logCtx.failed();
+                    return publisher.notifyErr(res, channel, providerConfigKey, connectionId, WSErrBuilder.UnknownError('failed to create connection'));
+                }
+
+                if (session.connectSessionId) {
+                    const connectSession = await getConnectSession(db.knex, {
+                        id: session.connectSessionId,
+                        accountId: account.id,
+                        environmentId: environment.id
+                    });
+                    if (connectSession.isErr()) {
+                        await logCtx.error('Failed to get session');
+                        await logCtx.failed();
+                        return publisher.notifyErr(res, channel, providerConfigKey, connectionId, WSErrBuilder.UnknownError('failed to get session'));
```

Comment:
same question about failing after the connection was successfully inserted in the db

Reply:
answered above